pkgs/by-name/pdfding/package.nix

```diff
@@ -0,0 +1,194 @@
+{
+  lib,
+  python3,
+  callPackage,
+  fetchFromGitHub,
+  fetchpatch2,
+  makeWrapper,
+}:
+let
+  python = python3.override {
+    self = python;
+    packageOverrides = final: prev: {
+      django = prev.django_5_2;
+    };
+  };
+
+  pythonPackages = with python.pkgs; [
+    django
+    django-allauth
+    django-cleanup
+    django-htmx
+    gunicorn
+    markdown
+    minio
+    nh3
+    psycopg2-binary
+    pypdf
+    pypdfium2
+    python-magic
+    qrcode
+    rapidfuzz
+    ruamel-yaml
+    whitenoise
+    huey
+    supervisor # required but not used by the module, using systemd instead
+    pillow
+    oauthlib
+
+    # dependecies required for django collectstatic
+    requests
+    pyjwt
+    cryptography
+  ];
+
+  frontend = callPackage ./frontend.nix { };
+
+  pythonPath = python.pkgs.makePythonPath pythonPackages;
+in
+
+python.pkgs.buildPythonApplication rec {
+  pname = "pdfding";
+  version = "1.4.1";
+  src = fetchFromGitHub {
+    owner = "mrmn2";
+    repo = "PdfDing";
+    tag = "v${version}";
+    hash = "sha256-8e80gMdB6U3977dIU7bIAAEguYmi+AWQgUgYPDLCYLI=";
+  };
+  pyproject = true;
+
+  patches = [
+    # remove in 1.4.2 (next version after 1.4.1)
+    # patch to add data_dir
+    # https://github.com/mrmn2/PdfDing/pull/202
+    (fetchpatch2 {
+      url = "https://github.com/mrmn2/PdfDing/commit/387ca2079f74844203e2e91fac00e0d0e0e5fdb9.patch?full_index=1";
+      hash = "sha256-VGjyIAVi+qd2WZ8FVKKC2ijLinoflO7RmPwIW1/oGcY=";
+    })
+    # pyproject.toml still has 0.1.1 very old version
+    # follow https://github.com/mrmn2/PdfDing/pull/203
+    ./add-version.patch
```

Comment:
No need to vendor the patch since it's been merged.

Reply:
We can rename the license in `postFetch` (so the patch could apply) and remove it after 1.4.2.

Reply:
done